packages/lesswrong/server/callbacks/commentCallbacks.ts

```diff
@@ -324,7 +324,7 @@ async function validateDeleteOperations (modifier, comment, currentUser) {
       if (
         (comment.deleted || comment.deletedPublic) &&
         (deletedPublic || deletedReason) &&
-        !Users.canDo('comments.remove.all') &&
+        !Users.canDo(currentUser, 'comments.remove.all') &&
```

Comment:
TODO: Figure out what consequence this type error had (the user was missing, making this always false I think), and whether the now-active code was otherwise okay. Same for the `comment.remove.all` check below (note "comment" vs "comments" inconsistency)